addons/stock/models/product.py

```diff
@@ -370,9 +379,15 @@ def _search_product_quantity(self, operator, value, field):
         ids = []
         # Order the search on `id` to prevent the default order on the product name which slows
         # down the search because of the join on the translation table to get the translated names.
-        for product in self.with_context(prefetch_fields=False).search([], order='id'):
+        products = self.with_context(prefetch_fields=False).search([('type', '!=', 'service')], order='id')
+        for product in products.with_context(skip_products_filter=True):
             if OPERATORS[operator](product[field], value):
                 ids.append(product.id)
+        # Add all services if 0.0 quantity is included in the result
+        if (value < 0.0 and operator in ('>', '>=') or
+            value > 0.0 and operator in ('<', '<=') or
+            value == 0.0 and operator in ('>=', '<=', '=')):
+            ids += self.search(['type', '=', 'service']).ids
```

Comment:
Do you really need this ? Couldn't you just do another search for 'service' type products and use the same if condition as above ?